cmd/memberagent/main.go

```diff
@@ -160,8 +160,23 @@ func buildHubConfig(hubURL string, useCAAuth bool, tlsClientInsecure bool) (*res
 
 	hubConfig.TLSClientConfig.Insecure = tlsClientInsecure
 	if !tlsClientInsecure {
-		hubConfig.TLSClientConfig.CAFile = os.Getenv("CA_BUNDLE")
-		hubCA := os.Getenv("HUB_CERTIFICATE_AUTHORITY")
+		caBundle, ok := os.LookupEnv("CA_BUNDLE")
+		if ok && caBundle == "" {
+			err := errors.New("environment variable CA_BUNDLE should not be empty")
+			klog.ErrorS(err, "failed to validate system variables")
```

Comment:
Why not return the err here? I am not sure why we just log this as user can't see the error log.  I don't think an empty env is valid.  I assume that not setting it means using os default but setting it as "" ambiguous.

Reply:
I think that is a copy/paste bug. I fixed that and added unit test in [this commit](https://github.com/Azure/fleet/pull/365/commits/f1f019903055963bbd77be13303157da388d600f).